packages/devtools_app/lib/src/profiler/cpu_profile_model.dart

```diff
@@ -407,3 +423,47 @@ int stackFrameIdCompare(String a, String b) {
     throw error;
   }
 }
+
+class CpuProfileStore {
+  final _store = <TimeRange, CpuProfileData>{};
+
+  CpuProfileData storedProfile(TimeRange time) {
+    // If we have a profile for a time range encompassing [time], then we can
+    // generate and cache the profile for [time] without needing to pull data
+    // from the vm service.
+    _maybeGenerateSubProfile(time);
+    return _store[time];
+  }
+
+  void storeProfile(TimeRange time, CpuProfileData profile) {
+    _store[time] = profile;
+  }
+
+  void _maybeGenerateSubProfile(TimeRange time) {
```

Comment:
nit: this is generating a bunch of overlapping profiles in the cache. That would make it a bit harder to later serialize this.   Is computing the sub profiles expensive enough that it worth it to add completely overlapping ranges to the cache. It makes sense we will need some overlap as  frames overlap but having completely overlapping entries seems a little odd.

Reply:
when a sub profile is generated, it has to be created (not expensive) and then processed - this is the expensive part. So if we don't cache each subprofile, there will be processing cost for each time we have to regenerate the sub profile from a super profile.